src/text-field-components.tsx

```diff
@@ -217,11 +140,23 @@ export const FieldContainer: React.FC<FieldContainerProps> = ({
     fullWidth,
     readOnly,
 }) => {
-    const classes = useFieldContainerStyles({multiline, fullWidth, disabled, readOnly});
-
     return (
-        <div className={classes.fieldContainer}>
-            <div className={classnames(classes.border, className)} ref={fieldRef}>
+        <div
+            className={classnames(styles.fieldContainer, fullWidth ? styles.fullWidth : styles.normalWidth, {
+                [styles.disabled]: disabled,
+            })}
+        >
+            <div
+                className={classnames(
+                    styles.field,
+                    multiline ? styles.fieldMulti : styles.fieldSingle,
+                    readOnly
+                        ? sprinkles({background: vars.colors.neutralLow})
+                        : sprinkles({background: vars.colors.backgroundContainer}),
```

Comment:
or `sprinkles({ background: readOnly ? vars.colors.neutralLow : vars.colors.backgroundContainer})`

IMO this way is clear to know that readOnly just applies to background property, but matter of taste

Reply:
ok